### DISCUSSION TITLE
Unified documentation of feature engineering (volunteering)

### DISCUSSION THREAD

**Discussion by willsmithorg** [autogluon/autogluon#1450] [Ideas]
I would like to help by writing a unified single page in the documentation covering all the feature engineering that AG does before calling the models.

Would this be welcome?  Some of it is described elsewhere in the documentation, but a lot of it is only discoverable from the source.

I'm thinking of a bit higher level than describing each generator in features/generators.

More at the level of :
- what is the universe of types each column can be
- how do we infer types per column
- for each type, what further feature engineering is done (e.g. ngrams and the date parsing I just submitted)
- what are the parameters used for the heuristics
- how does the user disable/enable bits of feature engineering.

Please let me know your thoughts.

**Comment by Innixma** [comment]
That sounds awesome! (Minor comment: Would recommend creating issues instead of discussions for contribution proposals, easier to track)

I actually created an example on FeatureGenerator awhile ago here: https://github.com/awslabs/autogluon/blob/master/examples/tabular/example_custom_feature_generator.py

I plan to rework it into a tutorial for v0.4 release (tracked in #1437)

However, this tutorial will not necessarily cover the same content as you are describing, so I don't think there is much overlap.

Please feel free to take a stab at it and let me know if you have any questions / design recommendations!

**Comment by willsmithorg** [comment]
I have raised #1456 for this.
